code/modules/integrated_electronics/subtypes/logic.dm

```diff
@@ -30,6 +30,16 @@
 /obj/item/integrated_circuit/logic/binary/proc/do_compare(var/datum/integrated_io/A, var/datum/integrated_io/B)
 	return FALSE
 
+/obj/item/integrated_circuit/logic/binary/proc/comparable(var/datum/integrated_io/A, var/datum/integrated_io/B)
+	if(isnum(A.data))
```

Comment:
this would read a bit clearer if it was formatted as `if (isnum(A.data) && isnum(B.data))`

Reply:
ditto for the `else if` branch